iib/web/models.py

```diff
@@ -728,11 +728,6 @@ def _from_json(
         if current_user.is_authenticated:
             request_kwargs['user'] = current_user
 
-        # Add the request to a new batch
-        batch = Batch()
```

Comment:
I don't understand why this was moved.

Reply:
The choice was between adding another arg to the `cls._from_json` function of the parent class right from the API call (when annotations are present) and creating a batch in the from_json methods of the respective child classes. I chose the latter because it now also becomes consistent with the from_json method of the RegenerateBundle class. 

If you feel strongly against it, I would be happy to move it back to where it was and introduce an extra arg.

Reply:
Hang on. All you have to do is add a new optional parameter, `batch`, to `RequestIndexImageMixin._from_json`, `RequestAdd.from_json`, and `RequestRm.from_json`. The last two already call the first. It's just then a matter of forwarding whatever `batch` parameter was given.

I don't have a strong opinion on this. I just don't think the duplication is necessary though.

Reply:
You got to me. I have made the change. :)